credentialdigger/cli/hook.py

```diff
@@ -0,0 +1,180 @@
+It detects hardcoded secrets in staged files.
+
+NOTE: It uses SQLite and the discoveries are saved (by default)
+in /home/USER/.local/data.db
+
+usage: credentialdigger hook [-h] [--dotenv DOTENV] [--rules RULES]
+                             [--db_path DB_PATH] [--no_interaction]
+
+optional arguments:
+  -h, --help         show this help message and exit
+  --dotenv DOTENV    The path to the .env file which will be used in all
+                     commands. If not specified, the one in the current
+                     directory will be used (if present).
+  --rules RULES      Specify the yaml file path containing the scan rules
+                     e.g., /path/to/rules.yaml
+  --db_path DB_PATH  Specify the database file path where to save the results
+                     e.g., /path/to/data.db
+  --no_interaction   Flag used to remove the interaction i.e.,
+                     do not prompt if the commit should continue
+                     in case of discoveries. If specified, the hook will
+                     fail in case of discoveries.
+"""
+
+import subprocess
+import sys
+from pathlib import Path
+
+from credentialdigger import SqliteClient
+from credentialdigger.models.model_manager import ModelManager
+
+
+def configure_parser(parser):
+    """ Configure arguments for command line parser.
+
+    Parameters
+    ----------
+    parser: `credentialdigger.cli.customParser`
+        Command line parser
+    """
+    parser.set_defaults(func=run)
+
+
+def system(*args, **kwargs):
+    """Run a command and get the result."""
+    kwargs.setdefault('stdout', subprocess.PIPE)
+    proc = subprocess.Popen(args, **kwargs)
+    out, err = proc.communicate()
+    return out
+
+
+def print_msg(msg):
+    """Print messages to /dev/tty"""
+    subprocess.run(f'echo \"\n{msg}\n\" > /dev/tty',
+                   shell=True,
+                   stdout=subprocess.PIPE)
+
+
+def ask_commit(str_discoveries):
+    """Ask for the commit confirmation in case of possible leaks"""
+
+    msg = 'You have the following disoveries:\n' \
+          f'{str_discoveries}\ncontinue? (y/n)'
+    print_msg(msg)
+
+    sys.stdin = open('/dev/tty', 'r')
+    # Create a process on /dev/tty to capture the input (commit or not)
+    # It reads the input, saves it in userinput and echos it
+    # subprocess.check_output return the output of the command i.e., userinpput
+    user_input = subprocess.check_output('read -p \"\" userinput && echo '
+                                         '\"$userinput\"',
+                                         shell=True, stdin=sys.stdin).rstrip()
+
+    return user_input.decode('utf-8')
+
+
+def run(args):
+    """Run credential digger on staged files
+
+    Parameters
+    ----------
+    args: `argparse.Namespace`
+        Arguments from command line parser.
+
+    Returns
+    -------
+        It exits with success code 0 if there are no discoveries, otherwise
+        it exits with an error code != 0 printing the leaks
```

Comment:
We have to be consistent with the return type. Do we return an int or an exit code? we cannot return an int (0) in a case and an exit code in another case. You can check the following example from the `scan` CLI:

```
While this function returns nothing of use to the scanner itself, it
gives an exit status (integer) that is equal to the number of
discoveries. If it exits with a value that is equal to 0, then it means
that the scan detected no leaks in this repo.
```

Maybe it's what happens also in this new CLI, but in this case you have to reformulate this docstring